hail/src/main/scala/is/hail/io/plink/LoadPlink.scala

```diff
@@ -44,14 +45,14 @@ object LoadPlink {
     """^-?(?:\d+|\d*\.\d+)(?:[eE]-?\d+)?$""".r
 
   def parseFam(filename: String, ffConfig: FamFileConfig,
-    fs: FS): (IndexedSeq[Row], TStruct) = {
+    fs: FS): (IndexedSeq[Row], PStruct) = {
 
     val delimiter = unescapeString(ffConfig.delimiter)
 
-    val phenoSig = if (ffConfig.isQuantPheno) ("quant_pheno", TFloat64()) else ("is_case", TBoolean())
+    val phenoSig = if (ffConfig.isQuantPheno) ("quant_pheno", PFloat64()) else ("is_case", PBoolean())
 
-    val signature = TStruct(("id", TString()), ("fam_id", TString()), ("pat_id", TString()),
-      ("mat_id", TString()), ("is_female", TBoolean()), phenoSig)
+    val signature = PStruct(("id", PString()), ("fam_id", PString()), ("pat_id", PString()),
```

Comment:
Not certain this change is worthwhile, because it's only used in Table, and immediately converted back to virtualType.

Reply:
This violates the no-ptypes-if-no-offheap-data principle, but I think this is probably right -- we'll want to make some of these fields required, when we start doing that everywhere, and we won't be able to do that on virtual types.